packages/synapse-react-client/src/components/useImmutableTableQuery.ts

```diff
@@ -211,6 +201,46 @@ export default function useImmutableTableQuery(
       ],
     )
 
+  /**
+   * Inspect the URL on mount to see if we have a particular query request that we must show.
+   */
+  useEffect(() => {
+    const queryRequestFromLink = DeepLinkingUtils.getQueryRequestFromLink(
+      'QueryWrapper',
+      componentIndex,
+    )
+    if (queryRequestFromLink && queryRequestFromLink.query) {
+      setQuery(prevState => ({
+        ...prevState,
+        ...queryRequestFromLink,
+        query: {
+          ...prevState.query,
+          ...queryRequestFromLink.query,
+        },
+      }))
+    }
+  }, [componentIndex])
+
+  // If `shouldDeepLink` is true, synchronize the URL
+  useEffect(() => {
+    if (shouldDeepLink) {
+      if (isEqual(initQueryRequest, lastQueryRequest)) {
+        DeepLinkingUtils.updateUrlWithNewSearchParam(
+          'QueryWrapper',
+          componentIndex,
+          null,
+        )
+      } else {
+        const queryJsonString = JSON.stringify(lastQueryRequest.query)
+        DeepLinkingUtils.updateUrlWithNewSearchParam(
+          'QueryWrapper',
+          componentIndex,
+          queryJsonString,
+        )
+      }
+    }
+  }, [componentIndex, lastQueryRequest, initQueryRequest, shouldDeepLink])
```

Comment:
Handle `shouldDeepLink` in this effect.

If the `initQueryRequest` matches the current query, pass `null` to  `updateUrlWithNewSearchParam`, which will cause the search param to be dropped.

Otherwise, update the URL as desired.

Reply:
Also, no longer use `encodeURIComponent` because the string is now passed to URLSearchParams, which handles encoding internally.

I have verified that an encoded string created before these changes can be decoded after these changes, so links in portals including an encoded query will not be broken.